url2f44efcfa6_wangyudong_sohutw_java_sdk_2_0_tgz-pJ8-com_sohu_t_api_sdk_json_TestJ8/src/com/sohu/t/api/sdk/data/Search.java

```diff
@@ -0,0 +1,155 @@
+package com.sohu.t.api.sdk.data;
+
+import com.sohu.t.api.sdk.TBlogException;
+import com.sohu.t.api.sdk.http.Response;
+import com.sohu.t.api.sdk.json.JSONArray;
+import com.sohu.t.api.sdk.json.JSONException;
+import com.sohu.t.api.sdk.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * User: chenxiaojian
+ * Date: 13-1-8
+ * Time: 下午12:31
+ */
+public class Search {
+    private String profile_image_url;
+    private String created_at;
+    private String from_user;
+    private long to_user_id;
+    private String text;
+    private long id;
+    private long from_user_id;
+    private String geo;
+    private String iso_language_code;
+    private String source;
+
+
+    public static List<Search> constructSearches(Response res) throws TBlogException {
+
+        return constructSearches(res.asJSONArray());
+    }
+
+
+    public static List<Search> constructSearches(JSONArray jsonArray) throws TBlogException {
+        try {
+            int size = jsonArray.length();
+            List<Search> searchUsers = new ArrayList<Search>(size);
+            for (int i = 0; i < size; i++) {
+                searchUsers.add(new Search(jsonArray.getJSONObject(i)));
+            }
+            return searchUsers;
+        } catch (JSONException jsone) {
+            throw new TBlogException(jsone);
+        } catch (TBlogException te) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)